components/mongodb/actions/create-new-document/create-new-document.mjs

```diff
@@ -0,0 +1,52 @@
+import mongodbApp from "../../mongodb.app.mjs";
+
+export default {
+  key: "mongodb-create-new-document",
+  name: "Create New Document",
+  description: "Create a new document in a collection of your choice. [See the docs here](https://docs.mongodb.com/manual/tutorial/insert-documents/)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    mongodbApp,
+    collection: {
+      propDefinition: [
+        mongodbApp,
+        "collection",
+      ],
+    },
+    data: {
+      label: "Data",
+      type: "object",
+      description: "The object to be used in document creation",
+    },
+    parseNumbers: {
+      label: "Parse Numbers",
+      type: "boolean",
+      description: "If `true`. All number values represented by a string will be parsed to it respective type",
+      default: true,
+    },
+    parseBooleans: {
+      label: "Parse Booleans",
+      type: "boolean",
+      description: "If `true`. All boolean values represented by a string will be parsed to it respective type",
+      default: true,
+    },
+    parseDates: {
+      label: "Parse Dates",
+      type: "boolean",
+      description: "If `true`. All date values represented by a string will be parsed to it respective type",
+      default: true,
+    },
+  },
```

Comment:
Since props `data`, `parseNumbers`, `parseBooleans`, and `parseDates` are used in multiple components, we should put them in the app file and use propDefinitions.